luomu-libpcap/src/error.rs

```diff
@@ -53,17 +54,36 @@ impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             Error::Break => write!(f, "libpcap: Loop terminated by pcap_breakloop (PCAP_ERROR_BREAK)."),
-            Error::NotActivated => write!(f, "libpcap: The capture needs to be activated (PCAP_ERROR_NOT_ACTIVATED)."),
-            Error::AlreadyActivated => write!(f, "libpcap: Capture handle already activated (PCAP_ERROR_ACTIVATED)."),
-            Error::NoSuchDevice => write!(f, "libpcap: The capture source specified when the handle was created doesn't exist (PCAP_ERROR_NO_SUCH_DEVICE)."),
-            Error::MonitorModeNotSupported => write!(f, "libpcap: Monitor mode was specified but the capture source doesn't support monitor mode (PCAP_ERROR_RFMON_NOTSUP)."),
-            Error::OnlySupportedInMonitorMode => write!(f, "libpcap: The operation is supported only in monitor mode (PCAP_ERROR_NOT_RFMON)."),
-            Error::PermissionDenied => write!(f, "libpcap: The process doesn't have permission to open the capture source (PCAP_ERROR_PERM_DENIED)."),
-            Error::InterfaceNotUp => write!(f, "libpcap: The capture source device is not up (PCAP_ERROR_IFACE_NOT_UP)."),
-            Error::TimestampTypeNotSupported => write!(f, "libpcap: This device doesn't support setting the time stamp type (PCAP_ERROR_CANTSET_TSTAMP_TYPE)."),
-            Error::PromiscuousPermissionDenied => write!(f, "libpcap: The process has permission to open the capture source but doesn't have permission to put it into promiscuous mode (PCAP_ERROR_PROMISC_PERM_DENIED)."),
-            Error::TimestampPrecisionNotSupported => write!(f, "libcap: The requested time stamp precision is not supported (PCAP_ERROR_TSTAMP_PRECISION_NOTSUP)."),
-
+            Error::NotActivated(interface) => {
+                write!(f, "libpcap: The capture handle {} needs to be activated (PCAP_ERROR_NOT_ACTIVATED).", interface)
+            }
+            Error::AlreadyActivated(interface) => {
+                write!(f, "libpcap: Capture handle {} already activated (PCAP_ERROR_ACTIVATED).", interface)
+            }
+            Error::NoSuchDevice(interface) => {
+                write!(f, "libpcap: The capture source {} when the handle was created doesn't exist (PCAP_ERROR_NO_SUCH_DEVICE).", interface)
```

Comment:
Maybe just "capture source {} does not exist", that would be much clearer.

Reply:
"Capture interface {} doesn't exist"